src/core/crypto/crypto_platform.cpp

```diff
@@ -789,6 +743,58 @@ OT_TOOL_WEAK void otPlatCryptoPbkdf2GenerateKey(const uint8_t *aPassword,
         key += useLen;
         keyLen -= useLen;
     }
+
+exit:
+    return error;
 }
 
 #endif // #if OPENTHREAD_FTD
+
+#elif OPENTHREAD_CONFIG_CRYPTO_LIB == OPENTHREAD_CONFIG_CRYPTO_LIB_PSA
```

Comment:
I'm wondering if we should also add `OT_TOOL_WEAK otError otPlatCryptoPbkdf2GenerateKey` here and return `OT_ERROR_NOT_IMPLEMENTED` ? This way we would make it clear that `otPlatCryptoPbkdf2GenerateKey`'s implementation using PSA should be done on platform's side
thoughts?

Reply:
Added an empty function, returning not capable error code.